fdbclient/MultiVersionTransaction.actor.cpp

```diff
@@ -1896,8 +1896,17 @@ void MultiVersionApi::loadEnvironmentVariableNetworkOptions() {
 			std::string valueStr;
 			try {
 				if (platform::getEnvironmentVar(("FDB_NETWORK_OPTION_" + option.second.name).c_str(), valueStr)) {
+					FDBOptionInfo::FDBOptionParamType curParamType = option.second.paramType;
+					int64_t intParamVal;
 					for (auto value : parseOptionValues(valueStr)) {
-						Standalone<StringRef> currentValue = StringRef(value);
+						Standalone<StringRef> currentValue;
+						if (curParamType == FDBOptionInfo::Int) {
+							intParamVal = std::stoll(value.c_str(),nullptr,10);
```

Comment:
`stoll` takes an `std::string`, so we don't need to call `.c_str()`. Additionally, because this variant throws exceptions we will need to catch them and convert to one of our `Error` types (e.g. `invalid_option_value`).

```suggestion
							try {
								intParamVal = std::stoll(value, nullptr, 10);
							} catch (std::exception e) {
								TraceEvent(SevError, "EnvironmentVariableParseIntegerFailed")
									.detail("Option", option.second.name)
									.detail("Value", valueStr)
									.detail("Error", e.what());
								throw invalid_option_value();
							}
```

Reply:
@sfc-gh-chliu We'll need to a change like the one above. Not sure why it got marked outdated.